### ISSUE TITLE
subscript and superscript support

### ISSUE THREAD

**Issue by benbro** [quilljs/quill#249]
subscript and superscript are useful for simple math and chemistry equations:
H<sub>2</sub>O
a<sup>2</sup> + b<sup>2</sup> = c<sup>2</sup>

HTML:
```html
<sub>subscript</sub>
<sup>superscript</sup>
```
Icons
http://fortawesome.github.io/Font-Awesome/icon/subscript/
http://fortawesome.github.io/Font-Awesome/icon/superscript/

Thanks

**Comment by benbro** [comment]
Are there plans to add subscript and superscript?
Is it possible to add support with a plugin?

**Comment by jhchen** [comment]
It is possible to add this with a plugin now as the normalizer has changed to be an instance with a variable list of allowed tags as opposed to a hard coded list but there is also a plan to add this officially and the normalizer change was one of the steps.

**Comment by benbro** [comment]
I've added superscript and subscript buttons to the master branch similar to the bold button.
It works in FF but Chrome is nesting subscript and superscript tags instead of toggling it on and off 
when calling
```javascript
execCommand('superscript')
```
When trying to get:
H<sub>2</sub>O
In Chrome I'm getting
H<sub>2<sub>O</sub></sub>

Does quill have a way to force Chrome to toggle the tag instead of nesting it?

**Comment by benbro** [comment]
Branch with superscript and subscript support:
https://github.com/benbro/quill/tree/superscript

To reproduce, please try to use subscript in Chrome, type several letters and than try to toggle it off by pressing the subscript button again. You'll get nested subscript tags.

When I'm trying to use execCommand('superscript') directly in Chrome it works as intended:
http://jsbin.com/fasobalocu/1/

**Comment by benbro** [comment]
Can you please explain how to add subscript and superscript as a plugin with the list of allowed tags?
Are you still planning to add this officially?

**Comment by benbro** [comment]
The problem with Chrome nesting sup and sub elements instead of toggling can be fixed with css:
```css
.ql-editor sup {
    vertical-align: super;
}
.ql-editor sub {
    vertical-align: sub
}
```
http://stackoverflow.com/questions/17517839/jquery-execcommand-for-superscript-and-subscript-not-properly-toggling

Not sure if that's a bug in Chrome or the expected behavior.

**Comment by benbro** [comment]
Can we prevent the user from using superscript and subscript at the same time?

**Comment by jhchen** [comment]
I'm still planning on adding this once the work on Custom Formats is done. That implementation will not rely on execCommand so existing formats will be much less buggy and new formats can be possible. I personally agree with not having both superscript and subscript at the same time and this would be a sensible default but this behavior will be very easy to control with Custom Formats.

**Comment by kmoe** [comment]
@benbro is your superscript branch still available? I'm probably going to have to implement super/subscript at some point so it would be useful to have a reference.

**Comment by benbro** [comment]
@kmoe sorry but no.
superscript, subscript and text-direction support is very important and common in text editors but it's very hard for me to keep patching quill in every version.

**Comment by kmoe** [comment]
Thanks - I'll let you know how I get on in implementing it myself (if we do go with using Quill).

**Comment by kmoe** [comment]
@benbro it seems to work fine here: https://github.com/quilljs/quill/compare/develop...kmoe:superscript-subscript
Testing in other browsers now, but I haven't seen the nested superscript tags yet.

**Comment by kmoe** [comment]
@benbro The problem has started happening again for some reason in Chrome, Opera, and Safari. I need to fix this so I'll update here when I have a solution or workaround.
As a last resort, we might try only enabling the toolbar when a user has text selected, as in that situation the superscript toggles fine.

**Comment by kmoe** [comment]
Ah, I reproduced it in the advanced example somehow.

**Comment by PavSidhu** [comment]
I tried using to change to superscript with quill however I saw on change. According to the documentation support for custom formats are not available yet
   document.execCommand('superscript');

**Comment by benbro** [comment]
Implemented in 1.0-beta.0
http://beta.quilljs.com/docs/formats/

Thanks
